### PR TITLE
Add Show-Info Button

### PR DESCRIPTION
**Pull request by oy-gen**
This pull request also includes a bigger file re-structure and design re-layout.

### CODE REVIEW

File: components/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import styled from 'styled-components';
+import QuantitySelector from './QuantitySelector';
+import { useState } from 'react';
+
+export default function ProductCard({ product }) {
+  const [showDetails, setShowDetails] = useState();
+
```
Comment (cat-miller) [inline components/ProductCard.js:8]:
```suggestion
const { id, image, name, ... } = product
```
consider to destructure

File: components/ProductCard.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  return (
+    <>
+      <StyledCard key={product.id}>
+        <StyledImage src={product.image} alt={product.name} />
```
Comment (cat-miller) [inline components/ProductCard.js]:
```suggestion
        <StyledImage src={product.image} alt="" />
```
in line 13 the name is already available for screen readers

File: components/ProductCard.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        <MainInfoWrapper>
+          <StyledProductName>{product.name}</StyledProductName>
+          <StyledWSPrice>WS {product.WSprice.toPrecision(4)} €</StyledWSPrice>
```
Comment (cat-miller) [inline components/ProductCard.js]: take a look at `toLocaleString`
there's an option for currency
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

File: components/ProductCard.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+          <MainInfoWrapperRow>
+            <StyledMoreInfoButton
```
Comment (cat-miller) [inline components/ProductCard.js]: why is your styled button a span?

File: components/ProductCard.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      </StyledCard>
+      {showDetails ? (
+        <ExtraInfoWrapper>
+          <StyledExtraInfo>
+            <strong>RRP: </strong> {product.RRPprice.toPrecision(4)} €
+            <br /> <strong>BARCODE:</strong> {product.id}
+          </StyledExtraInfo>
+          <StyledExtraInfo>{product.description}</StyledExtraInfo>
+        </ExtraInfoWrapper>
+      ) : null}
```
Comment (cat-miller) [inline components/ProductCard.js]:
```suggestion
      {showDetails && (
        <ExtraInfoWrapper>
          <StyledExtraInfo>
            <strong>RRP: </strong> {product.RRPprice.toPrecision(4)} €
            <br /> <strong>BARCODE:</strong> {product.id}
          </StyledExtraInfo>
          <StyledExtraInfo>{product.description}</StyledExtraInfo>
        </ExtraInfoWrapper>
      )}
```

File: components/ProductCard.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  align-items: center;
+  border-top: 1px solid lightgray;
+  background-color: var(--background-color);
+  padding: 0.8rem;
+`;
+const MainInfoWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  flex-grow: 1;
```
Comment (cat-miller) [inline components/ProductCard.js]: this should be a grid.

File: components/ProductCard.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  gap: 0.6rem;
+  padding-left: 0.6rem;
+`;
+
+const MainInfoWrapperRow = styled.div`
```
Comment (cat-miller) [inline components/ProductCard.js]: the name is a bit confusing, probably it should call FlexWrapper?

File: components/ProductCard.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const ExtraInfoWrapper = styled.div`
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
```
Comment (cat-miller) [inline components/ProductCard.js]: are you sure, that this lines take effect, if it's not a `display: flex`?
i think you don't need it

File: components/ProductCard.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  padding: 0.8rem;
+  max-width: 600px;
+  margin: auto;
+`;
+
+const StyledProductName = styled.h2`
+  font-family: 'Poppins', sans-serif;
```
Comment (cat-miller) [inline components/ProductCard.js]: consider styling general headline attributes inside the body.

File: components/ProductCard.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+  color: var(--text-maincolor);
+  font-size: 1rem;
+  font-weight: 200;
+  letter-spacing: 0.4rem;
+  text-transform: uppercase;
```
Comment (cat-miller) [inline components/ProductCard.js]: consider styling general headline attributes inside the body.

File: components/ProductCard.js [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+  @media (max-width: 600px) {
+    font-size: 14px;
```
Comment (cat-miller) [inline components/ProductCard.js]: please stay with rem

File: components/ProductCard.js [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+  }
+`;
+
+const StyledWSPrice = styled.h3`
+  font-family: 'Poppins', sans-serif;
+  text-transform: uppercase;
```
Comment (cat-miller) [inline components/ProductCard.js]: consider styling general headline attributes inside the body.

File: components/ProductCard.js [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+  color: var(--text-maincolor);
+  font-size: 1rem;
```
Comment (cat-miller) [inline components/ProductCard.js]: consider styling general headline attributes inside the body.

File: components/ProductCard.js [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+  font-weight: 600;
+`;
+
+const StyledExtraInfo = styled.p`
+  font-family: 'Poppins', sans-serif;
```
Comment (cat-miller) [inline components/ProductCard.js]: if you use this font everywhere, consider to put it on the body inside global styles.

File: components/ProductCard.js [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+  padding-bottom: 0.4rem;
+  color: var(--text-lightcolor);
+  font-size: 14px;
```
Comment (cat-miller) [inline components/ProductCard.js]: You can even set the base font-size inside the body

File: components/ProductCard.js [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+            <StyledMoreInfoButton
+              type="button"
+              key={product.id}
+              onClick={() => {
+                setShowDetails(!showDetails);
+              }}
+            >
+              {showDetails ? 'Less Info' : 'More Info'}
+            </StyledMoreInfoButton>
+            <QuantitySelector product={product} />
```
Comment (cat-miller) [inline components/ProductCard.js]: consider just passing quantity and id instead of everything.

**Review by cat-miller — COMMENTED** [review]
For me you should avoid using flex-direction: column and consider styling your text appearance inside the GlobalStyles. That would make your css much more lightweight. 🥷

File: components/ProductCard.js (hunk 2 above)
Comment (oy-gen) [reply components/ProductCard.js]: as far as I know alt names also have SEO relevance. I would keep it.

File: components/ProductCard.js (hunk 3 above)
Comment (oy-gen) [reply components/ProductCard.js]: Nice! Thank you!

**Pull request merged**